R/cube-index-table.R

```diff
@@ -1,13 +1,14 @@
 #' Calculate an index table for a CrunchCube
 #'
 #' Index tables are percentages of percentages. They take the percentage from
-#' `prop.table()` and divide that by the proportions of the other margin.
+#' `prop.table(cube, margin)` and, by default, divide that by the proportions of the other margin. The `baseline` argument can be used to provide baseline proportions to compare against.
 #'
 #' `index.table()` is only implemented for 2 dimensional cubes. If you need to
 #' calculate indexes for a higher dimension Cube, please slice the cube first.
 #'
 #' @param x A CrunchCube to calculate index table for
 #' @param margin which margin to index against (1 for rows, 2 for columns)
+#' @param baseline an arbitrary set of proportions to compare the table given in `x` to. Useful for comparing two separate cubes. `baseline` must have the same length as the extent of the dimension given in `margin`.
```

Comment:
What's the use case for having this argument?

Reply:
The one I can think of now is if your cube's margin doesn't reflect the population you're interested in. The classic case we've been hearing is that you want to index to a nationally representative sample, so index with our default baseline will be that as long as the cube is weighted to reflect that sample. If the cube is unweighted or weighted to some other sample, the margin might not be the right baseline.

I also find it easier to reason about and experiment with to see what's going on to be able to provide my own baseline and see how that changes the values coming out.